### PR TITLE
Add backend support for specifying language version

### LINKED ISSUE #54: Feature: Add language versioning to the back-end
**Description:**
There will be some differences between different versions of languages. One example is Python 2 vs 3. They are similar but also very different. Someone should be able to choose a language but also a version to compare using the Code Thesaurus system.

**Requirements:**
* When the user requests a specific version of a language, the version should be reflected in the language that gets passed into the template.
* When submitted from the front-end (#53), the view functions should know what language they picked up
* The thesaurus files should be moved to subdirectories reflecting their version numbers
* The Compare and Reference view functions should correctly pick up the correct files from those version

**Notes:**
While the front-end issue (#53) is separate from this back-end issue, they will have to be merged in together.

**Claiming This Issue:**
* Please leave a comment below indicating your interest. I can then assign it to you.
* If I don't see an update (comment, code, or PR), then I'll take your name off.
* You would always be welcome to reclaim it again.
* If you can't work on it, either leave a comment or remove yourself from the issue.
* If you have questions, please let me know!

### PR DESCRIPTION
**Pull request by jwalter**
If no version is provided the first available language versions is selected. Closes #54 

(I'm a Python newbie so if anything looks funky I really want to know)

### CODE REVIEW

File: web/thesauruses/meta_info.json
Comment (geekygirlsarah) [inline web/thesauruses/meta_info.json:5]: I hadn't thought of an array of version numbers. Good idea!

**Comment by geekygirlsarah** [comment]
I like it! I'll merge it in (after I fix this merge conflict).

Thanks and have a great weekend!

**Comment by geekygirlsarah** [comment]
Hey @jwalter ! I like what you did here and I think it'll work great. I'll tag it as approved for Hacktoberfest, but can't merge it yet until the frontend ticket is done with it. So it should still count!

Thanks and have a great weekend!

**Comment by geekygirlsarah** [comment]
@jwalter Can you leave a comment on the issue (#54 ) so I can assign you to it? I don't want anyone else to try to grab it since it's already done. Thanks!

**Comment by jwalter** [comment]
> Hey @jwalter ! I like what you did here and I think it'll work great. I'll tag it as approved for Hacktoberfest, but can't merge it yet until the frontend ticket is done with it. So it should still count!
> 
> Thanks and have a great weekend!

I was trying to make the backend changes be compatible with the current frontend version. Did you find any problems I should fix?

**Comment by geekygirlsarah** [comment]
Oops, thanks for your patience on this.

When I view the PR deployment (https://codethesauru-backend-su-8b7oxi.herokuapp.com/ ), the pages don't pull up. They're expecting a version number to be passed in. So the backend looks solid but the current front end doesn't cooperate with it. Issue #53 was to work with this issue to get the front end to be compatible with this. This looks like it will work, and it's why I marked it good for Hacktoberfest and 👍 it but can't merge it yet because it'll break the current site.

If someone doesn't grab the front end ticket, I'll probably try to tackle it in the next week or two now that I'm close to getting a documentation site up and going.

**Review by geekygirlsarah — APPROVED** [review]

**Comment by geekygirlsarah** [comment]
As #53 was supposed to get implemented with this and never was, and the project's evolved over time and got resolved another way, and this now has numerous conflicts at this point, I'm going to go ahead and close it.

I want to say I do appreciate the effort you put into it, and it was definitely valid in my mind for Hacktoberfest and for Code Thesaurus, just the other piece never got done since you made this in October.

I hope you consider contributing in the future if you want!

**Pull request closed without merging**
